src/core/math_func.cpp

```diff
@@ -45,7 +45,8 @@ int GreatestCommonDivisor(int a, int b)
 		b = a % b;
 		a = t;
 	}
-	return a;
+	/* GCD is always positive */
+	return abs(a);
```

Comment:
Not sure this is the correct solution - maybe there should be asserts at the beginning that check for non negative numbers?

Reply:
If you intend limit the domain to positive numbers, then absolutely we could add some asserts at the beginning.

It seems that calculating the GCD of negative numbers in general does produce a positive result:
```
$ python3
Python 3.7.2+ (default, Feb  2 2019, 14:31:48) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import math
>>> math.gcd(-10, -25)
5
>>> math.gcd(10, -25)
5
```

Makes sense intuitively, since you can divide -25 by 5 without a remainder, and 5 > -5 of course.

Reply:
this should be correct, as if n divides both a and b, then -n does as well.

however, i would probably put the abs() at the beginning of the function to avoid any odd side effects in the algorithm